crates/bws/src/command/secret.rs

```diff
@@ -0,0 +1,166 @@
+    pub(crate) note: Option<String>,
+    pub(crate) project_id: Uuid,
+}
+
+#[derive(Debug)]
+pub(crate) struct SecretEditCommandModel {
+    pub(crate) id: Uuid,
+    pub(crate) key: Option<String>,
+    pub(crate) value: Option<String>,
+    pub(crate) note: Option<String>,
+    pub(crate) project_id: Option<Uuid>,
+}
+
+pub(crate) async fn list(
+    mut client: Client,
+    organization_id: Uuid,
+    project_id: Option<Uuid>,
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let res = if let Some(project_id) = project_id {
+        client
+            .secrets()
+            .list_by_project(&SecretIdentifiersByProjectRequest { project_id })
+            .await?
+    } else {
+        client
+            .secrets()
+            .list(&SecretIdentifiersRequest { organization_id })
+            .await?
+    };
+
+    let secret_ids = res.data.into_iter().map(|e| e.id).collect();
+    let secrets = client
+        .secrets()
+        .get_by_ids(SecretsGetRequest { ids: secret_ids })
+        .await?
+        .data;
+    serialize_response(secrets, output_settings.output, output_settings.color);
+
+    Ok(())
+}
+
+pub(crate) async fn get(
+    mut client: Client,
+    secret_id: Uuid,
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let secret = client
+        .secrets()
+        .get(&SecretGetRequest { id: secret_id })
+        .await?;
+    serialize_response(secret, output_settings.output, output_settings.color);
+
+    Ok(())
+}
+
+pub(crate) async fn create(
+    mut client: Client,
+    organization_id: Uuid,
+    secret: SecretCreateCommandModel,
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let secret = client
+        .secrets()
+        .create(&SecretCreateRequest {
+            organization_id,
+            key: secret.key,
+            value: secret.value,
+            note: secret.note.unwrap_or_default(),
+            project_ids: Some(vec![secret.project_id]),
+        })
+        .await?;
+    serialize_response(secret, output_settings.output, output_settings.color);
+
+    Ok(())
+}
+
+pub(crate) async fn edit(
+    mut client: Client,
+    organization_id: Uuid,
+    secret: SecretEditCommandModel,
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let old_secret = client
+        .secrets()
+        .get(&SecretGetRequest { id: secret.id })
+        .await?;
+
+    let new_secret = client
+        .secrets()
+        .update(&SecretPutRequest {
+            id: secret.id,
+            organization_id,
+            key: secret.key.unwrap_or(old_secret.key),
+            value: secret.value.unwrap_or(old_secret.value),
+            note: secret.note.unwrap_or(old_secret.note),
+            project_ids: match secret.project_id {
+                Some(id) => Some(vec![id]),
+                None => match old_secret.project_id {
+                    Some(id) => Some(vec![id]),
+                    None => bail!("Editing a secret requires a project_id."),
```

Comment:
I think it would be cleaner to return an Err here and have the process exit handling in `main.rs`.

Reply:
Updated in https://github.com/bitwarden/sdk/pull/809/commits/f95d762d0399566135e5adb16fdd4625212b89d2.